metadata_backend/api/middlewares.py

```diff
@@ -50,6 +53,78 @@ async def http_error_handler(req: Request, handler: Callable) -> Response:
+    """Middleware for validating and authenticating JSON web token.
+
+    :param req: A request instance
+    :param handler: A request handler
+    :raises: HTTP Exception with status code 401 or 403
+    :returns: Successful requests unaffected
+    """
+    @middleware
+    async def jwt_authentication(req: Request, handler: Callable) -> Response:
+        if 'Authorization' in req.headers:
+            # Check token exists
+            try:
+                scheme, token = req.headers.get('Authorization').split(' ')
+                LOG.info('Auth token received.')
+            except Exception as err:
+                raise web.HTTPUnauthorized(reason=str(err))
+
+            # Check token has proper scheme and was provided.
+            if not re.match('Bearer', scheme):
+                raise web.HTTPUnauthorized(reason="Invalid token scheme, "
+                                                  "Bearer required.")
+            if token is None:
+                raise web.HTTPUnauthorized(reason='Token cannot be empty.')
+
+            # JWK and JWTClaims parameters for decoding
+            key = environ.get('PUBLIC_KEY', None)
+            # TODO: get Oauth2 public key
+            # TODO: verify audience claim
+
+            # Include claims that are required to be present
+            # in the payload of the token
+            claims_options = {
+                "iss": {
+                    "essential": True,
+                    # TODO: add the proper issuer URLs here
+                    "values": ["haka_iss", "elixir_iss"]
+                },
+                "exp": {
+                    "essential": True
+                }
+            }
+
+            # Decode and validate token
+            try:
+                claims = jwt.decode(token, key, claims_options=claims_options)
+                claims.validate()
+                LOG.info('Auth token decoded and validated.')
+
+                # TODO: Retrieve GA4GH passports and process into permissions
+
+                req["token"] = {"authenticated": True}
+                return await handler(req)
+            except errors.MissingClaimError as err:
+                raise web.HTTPUnauthorized(reason=f"{err}")
+            except errors.ExpiredTokenError as err:
+                raise web.HTTPUnauthorized(reason=f"{err}")
+            except errors.InvalidClaimError as err:
+                raise web.HTTPForbidden(reason=f"Token contains {err}")
+            except errors.BadSignatureError as err:
+                raise web.HTTPUnauthorized(reason="Token signature is invalid"
+                                                  f", {err}")
+            except errors.InvalidTokenError as err:
+                raise web.HTTPUnauthorized(reason="Invalid authorization token"
+                                                  f": {err}")
+        else:
+            req["token"] = {"authenticated": False}
+            return await handler(req)
+
+    return jwt_authentication
```

Comment:
no need to return this, if you import jwt_authentication in servers.py (see previous comment)